cpp/src/arrow/engine/substrait/expression_internal.cc

```diff
@@ -138,6 +138,77 @@ std::string EnumToString(int value, const google::protobuf::EnumDescriptor* desc
   return value_desc->name();
 }
 
+Result<compute::Expression> FromProto(const substrait::Expression::ReferenceSegment* ref,
+                                      const ExtensionSet& ext_set,
+                                      const ConversionOptions& conversion_options,
+                                      std::optional<compute::Expression> in_expr) {
+  auto in_ref = ref;
+  auto& out = in_expr;
```

Comment:
This is strange - why are we talking `in_expr` as out?

Reply:
Fixed.